vm.c

```diff
@@ -0,0 +1,22 @@
+#include "vm.h"
+#include "libkern.h"
+
+/* This is the pointer to the extended end of kernel's image. See
+   vm_early_alloc for details. */
+char* vm_kernel_image_end;
```

Comment:
I was slightly hesitant about it. I suspected that the memory manager which will deal with the entire RAM pool `[bss_end, stack_end]` will need to know how much memory was allocated this way, and it might be implemented in a different module.

However, instead of peeking at `vm_kernel_image_end` it could as well allocate a 0-size block. So probably there is no need to expose this variable to different TU's.